File: 220819-dojo/javascript/solution.js
export default function solution(password, length) {
  const upperCases = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
  const lowerCases = 'abcdefghijklmnopqrstuvwxyz';

  const caesarPassword = password.split('')
    .map((word) => {
      const upperIndex = upperCases.indexOf(word);

      if (upperIndex !== -1) {
        return upperIndex + length <= 25
          ? upperCases[upperIndex + length]
          : upperCases[upperIndex + length - 26];
      }

      const lowerIndex = lowerCases.indexOf(word);

      if (lowerIndex !== -1) {
        return lowerIndex + length <= 25
          ? lowerCases[lowerIndex + length]
          : lowerCases[lowerIndex + length - 26];
      }

      return ' ';
    });

  return caesarPassword.join('');
}

File: 220819-dojo/java/app/src/main/java/Solution.java
import java.util.Arrays;
import java.util.stream.Collectors;

public class Solution {
  public String solution(String s, int n) {
    String answer = "";
    String alphabetCase = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"; // 0~25, 26~51
//    String lowerCase = "abcdefghijklmnopqrstuvwxyz";

    String[] words = s.split("");

    for (int i = 0; i < words.length; i += 1) {
      if (words[i].equals(" ")) {
        continue;
      }

      int index = alphabetCase.indexOf(words[i]);

      if (index >= 0 && index <= 25) {
        words[i] = alphabetCase.charAt((index + n) % 25) + "";
      }

      if (index >= 26 && index <= 51) {
        if (index + n <= 51) {
          words[i] = alphabetCase.charAt(index + n)+ "";
        }

        words[i] = alphabetCase.charAt(index + n - 26) + "";
      }
    }

    answer = Arrays.stream(words).collect(Collectors.joining(""));

    return answer;
  }
}
